lib/tapioca/runtime/trackers/required_ancestor.rb

```diff
@@ -10,18 +10,18 @@ module RequiredAncestor
         class << self
           extend T::Sig
 
-          sig { params(requiring: T::Helpers, block: T.proc.returns(Module)).void }
+          sig { params(requiring: T::Helpers, block: T.proc.void).void }
```

Comment:
Ugh, yeah it shouldn't be `void` upstream either, since that means the block returns nothing and only has side-effects, whereas, the only reason the block exists is to returns _something_.

In any case, as I said above, this is Tapioca code now, and not Sorbet runtime. Regardless of what upstream says about the signature, this code is assuming it returns something (since it calls the block and returns its return value) and thus should be typed properly, so that the rest of our code can make the proper assumptions about it.

Reply:
I just paired with @egiurleo on this. Unfortunately we couldn't find a way to declare the type for `T.class_of` as it's not simply `T::Types::ClassOf`. And the concept may only exist in Sorbet internal as `Runtime object representing type: T.class_of(Foo)`. 

I'll try to pair with @Morriar to see 1) if that's true, and 2) if it's possible to expose it to Ruby so we can declare it.